### PR TITLE
Pedersen commitments over BLS12-381

### PR DESCRIPTION
**Pull request by lovesh**

### CODE REVIEW

File: src/commitments/pedersen_BLS12_381.rs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        let mut blinding_factor = random_big_number(&CURVE_ORDER, None);
+        commitment.add(&gen.mul(&blinding_factor));
+
+        blinding_factor.tobytes(&mut blinding_factor_bytes);
+
+        // Zero the bytes of blinding factor.
+        // TODO: A better mechanism is needed. Use either libsodium or some other library that makes sure that `blinding_factor`
+        // was never swapped to disk or was core dumped.
+        blinding_factor.zero();
+
```
Comment (mikelodder7) [inline src/commitments/pedersen_BLS12_381.rs]: sodium_mlock I believe does this and is exposed with libsodium-ffi.
See https://download.libsodium.org/doc/helpers/memory_management.html

Comment (lovesh) [reply src/commitments/pedersen_BLS12_381.rs]: I know, but i want to explore a more thorough mechanism (or library) and implement the mechanism at one place and use it everywhere. The ones on my list are [clear_on_drop](https://github.com/cesarb/clear_on_drop) and [secrets](https://github.com/stouset/secrets)

**Comment by lovesh** [comment]
@mikelodder7 Regarding "Do we need Pedersen specifically for BLS-381?", the group operations are specific to the curve. Do you mean creating a generic over a `Group` where `Group` consists of G1 (ECP), G2 (ECP2)?

**Comment by mikelodder7** [comment]
The generic interface should pass those as inputs.

**Comment by lovesh** [comment]
@mikelodder7 The hash function depends on the group being used

**Pull request merged**